### DISCUSSION TITLE
Using supervisor for laravel queue worker

### DISCUSSION THREAD

**Discussion by insign** [shinsenter/php#158] [Q&A]
In order to avoid mess with the image, what is the best way to  configure a supervisor for laravel queue worker

**Comment by shinsenter** [comment]
@insign
I think you can use cron jobs
https://github.com/shinsenter/php?tab=readme-ov-file#using-cron-jobs

**Comment by insign** [comment]
I'm considering change the [supervisor env command](https://github.com/shinsenter/php?tab=readme-ov-file#customize-supervisor-command) since I'll use the same image but start a exclusive container for the worker, any objection in your opinion?

**Reply by shinsenter** [reply]
@insign 

Yes, I think that's also a viable option if you have the resources to run multiple containers for your project.

**Reply by insign** [reply]
Can I set a custom supervisor file?

**Reply by shinsenter** [reply]
@insign 

Yes, although I haven't documented it in detail, my Docker images containing the web server use [s6-overlay](https://github.com/just-containers/s6-overlay?tab=readme-ov-file#writing-a-service-script) to act as a supervisor for processes. 

I've also created a wrapper command `s6-service` to quickly set up a service using s6-overlay. 

Here is an example of how to create a service in your Dockerfile:

```dockerfile
# create s6-overlay services
if has-cmd s6-service; then
    s6-service laravel-queue longrun '#!/usr/bin/env sh
export APP_PATH="$(app-path)"
export APP_ROOT="$(app-root)"
cd $APP_PATH && exec php $APP_PATH/artisan queue:work
'
fi
```
